src/components/scenes/CryptoExchangeQuoteProcessingScene.tsx

```diff
@@ -47,9 +47,15 @@ export function CryptoExchangeQuoteProcessingScene(props: Props) {
   const { route, navigation } = props
   const { swapRequest, swapRequestOptions, onCancel, onDone } = route.params
 
+  const fromCurrencyCode = getCurrencyCode(swapRequest.fromWallet, swapRequest.fromTokenId)
+  const toCurrencyCode = getCurrencyCode(swapRequest.toWallet, swapRequest.toTokenId)
+
   const dispatch = useDispatch()
   const account = useSelector(state => state.core.account)
 
+  const fromDenomination = useSelector(state => selectDisplayDenomByCurrencyCode(state, swapRequest.fromWallet.currencyConfig, fromCurrencyCode))
+  const toDenomination = useSelector(state => selectDisplayDenomByCurrencyCode(state, swapRequest.toWallet.currencyConfig, toCurrencyCode))
+
```

Comment:
If you use the new `selectDisplayDenom` or even `useDisplayDenom` hooks, those take the tokenId natively, which helps with our currency code elimination.